### PR TITLE
build: create helm chart

### PR DESCRIPTION
**Pull request by iamniting**
Signed-off-by: Nitin Goyal <nigoyal@redhat.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -175,6 +175,12 @@ deploy: update-mgr-env manifests kustomize ## Deploy controller to the K8s clust
 	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+helm-chart: update-mgr-env manifests kustomize ## Deploy controller to the K8s cluster specified in ~/.kube/config.
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG} && $(KUSTOMIZE) edit set nameprefix ${MANAGER_NAME_PREFIX}
+	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
+	sed -i 's/^version.*/version: ${VERSION}/g' mychart/Chart.yaml
+	$(KUSTOMIZE) build config/default > mychart/templates/template.yaml
```
Comment (Madhu-1) [inline Makefile:182]: should rename mychart to charts? it's a common name mostly used

Comment (iamniting) [reply Makefile]: renamed it to charts

File: Makefile [hunk 2]
```diff
@@ -175,6 +175,13 @@ deploy: update-mgr-env manifests kustomize ## Deploy controller to the K8s clust
 	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
 	$(KUSTOMIZE) build config/default | kubectl apply -f -
 
+helm-chart: update-mgr-env manifests kustomize ## Create helm chart
+	cd config/manager && $(KUSTOMIZE) edit set image controller=${IMG} && $(KUSTOMIZE) edit set nameprefix ${MANAGER_NAME_PREFIX}
+	cd config/default && $(KUSTOMIZE) edit set image rbac-proxy=$(RBAC_PROXY_IMG)
+	$(KUSTOMIZE) build config/default > charts/templates/template.yaml
```
Comment (Madhu-1) [inline Makefile:181]: should we use `kustomize build --enable-helm` to build helm charts?

File: charts/values.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+ingress:
+  enabled: false
+  className: ""
+  annotations: {}
+    # kubernetes.io/ingress.class: nginx
+    # kubernetes.io/tls-acme: "true"
+  hosts:
+    - host: chart-example.local
+      paths:
+        - path: /
```
Comment (Madhu-1) [inline charts/values.yaml:52]: this PR is only to add the example templates?

File: Makefile (hunk 2 above)
Comment (leelavg) [reply Makefile:181]: that's cool, how much effort is required to edit final charts created by kustomize from any prior experience?

Comment (Madhu-1) [reply Makefile:181]: not sure about it, i just came to know that we can also build helm from kustomize tool https://tech.aabouzaid.com/2020/09/3-ways-to-customize-off-the-shelf-helm-charts-with-kustomize-kubernetes.html#method-1-kustomize-plugins https://github.com/kubernetes-sigs/kustomize/blob/master/examples/chart.md#helm-related-flags

File: charts/Chart.yaml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+# This is the chart version. This version number should be incremented each time you make changes
+# to the chart and its templates, including the app version.
+# Versions are expected to follow Semantic Versioning (https://semver.org/)
+version: 0.0.1
+
+# This is the version number of the application being deployed. This version number should be
+# incremented each time you make changes to the application. Versions are not expected to
+# follow Semantic Versioning. They should reflect the version the application is using.
+# It is recommended to use it with quotes.
+appVersion: "1.16.0"
```
Comment (leelavg) [inline charts/Chart.yaml:24]: - i think this'll show up in `helm ls` better have this same as `version` just a thought.

**Comment by iamniting** [comment]
closing it as we are not interested anymore in this PR.

**Pull request closed without merging**